testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractGradleBuildIsolatedProjectsSmokeTest.groovy

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+import org.gradle.initialization.StartParameterBuildOptions
+
+/**
+ * Smoke test building gradle/gradle with isolated projects enabled.
+ */
+abstract class AbstractGradleBuildIsolatedProjectsSmokeTest extends AbstractGradleBuildConfigurationCacheSmokeTest {
```

Comment:
❓ `AbstractGradleBuildConfigurationCacheSmokeTest` has some preconditions on it. Will it be inherited for this class? I see `AbstractGradleceptionSmokeTest` and `AbstractGradleBuildConfigurationCacheSmokeTest` have duplicated set of preconditions despite the single hierarchy, what is make me think that preconditions may be not a subject for inheritance.